### ISSUE TITLE
viewer.text_overlay.text

### ISSUE THREAD

**Issue by JosephBendahan** [napari/napari#5599]
Hi,
What is the name of the methods to change the color and set the position of the viewer.text_overlay.text?
BTW, viewer.text_overlay.color = 'red' crashes napari.
Better yet, is there doc to find all the methods?
Thx.

**Comment by psobolewskiPhD** [comment]
In principle the docs should all be on napari.org under API reference.
However, I see that this feature isn't documented! 🤔 

I don't believe that it is possible to move the text_overlay at the moment.

However, changing the color should work.
For me the following does work:
```
viewer.text_overlay.text = "Hello World"
viewer.text_overlay.visible = True
viewer.text_overlay.color = 'red'
```

If that is crashing napari for you, can you post the output of `napari --info` (or use the Help menu)

**Comment by JosephBendahan** [comment]
psobolewskiPhD),
My mistake, I used 'Red" with capital R. But, it should've not crashed napari.
Also, there should be documentation.

I found in text_overlay.py the following
    visible: bool = False
    color: ColorValue = (0.5, 0.5, 0.5, 1.0)
    font_size: float = 10
    position: TextOverlayPosition = TextOverlayPosition.TOP_LEFT
    text: str = ""
So, the position can be changed but for selected places, found in _viewer_constansts.py
    TOP_LEFT = 'top_left'
    TOP_RIGHT = 'top_right'
    TOP_CENTER = "top_center"
    BOTTOM_RIGHT = 'bottom_right'
    BOTTOM_LEFT = 'bottom_left'
    BOTTOM_CENTER = "bottom_center"
I guess if there is no documentation, we need to find the details in the source code.
Thank you.

**Comment by psobolewskiPhD** [comment]
Thanks for the followup! I don't get why the API reference on napari.org doesn't have this info, everything does have proper doc strings. Will look into getting that fixed.
Also, I've updated an existing issue that we need some tutorial or guide to overlays in general:
https://github.com/napari/docs/issues/37

Regarding the crash, can you clarify that a bit more? For me the following doesn't cause a crash, just `KeyError: 'Red'`
```
viewer.text_overlay.text = "Hello World"
viewer.text_overlay.visible = True
viewer.text_overlay.color = 'Red'		# Note the capital, which is not correct on purpose
```
This is with napari 0.4.17
Does this cause a crash for you?

Edit: reading the traceback, I think even the KeyError being returned may be a vispy bug:
```
File ~/Dev/miniforge3/envs/apple-TF210/lib/python3.9/site-packages/vispy/color/color_array.py:24, in _string_to_rgb(color)
     22     if color.lower() not in _color_dict:
     23         raise ValueError('Color "%s" unknown' % color)
---> 24     color = _color_dict[color]
     25     assert color[0] == '#'
     26 # hex color

KeyError: 'Red'
```
It checks the lower case in the color_dict, but then doesn't act on it...

**Comment by psobolewskiPhD** [comment]
Regarding the API docs, I think it's because the overlays were not added here:
https://github.com/napari/napari/blob/main/napari/components/__init__.py
So the auto-built docs don't pick up on them.
@brisvag were the overlays still intended to be hidden?

**Comment by brisvag** [comment]
Yeah, the overlay models are in abit of a limbo at the moment between private/public api, unfortunately. But maybe the pre-existing ones like text-overlay and axes should/could be made public.

> My mistake, I used 'Red" with capital R. But, it should've not crashed napari.

Oops, at leas that should be easy to solve with a `lower()`.

> So, the position can be changed but for selected places, found in _viewer_constansts.py

Correct; as you've seen, this part is not really fully public api, so that's why it's not well documented or discoverable. So thanks for trying it out and reporting issues!

**Comment by JosephBendahan** [comment]
I tried to reproduce the 'Red' but it does not happen now.
Thank you all for you responses!

**Comment by psobolewskiPhD** [comment]
> But maybe the pre-existing ones like text-overlay and axes should/could be made public.

I think this would be a good idea... lets add scale_bar to the list too.

**Comment by brisvag** [comment]
So it would entail just adding to the `__init__.__all__`? Not very familiar with the docs generation :P

**Comment by psobolewskiPhD** [comment]
I think so, based on the current docs. Maybe @melissawm can gives us a pointer or two.

**Comment by melissawm** [comment]
@brisvag in this case, adding a docstring to [text_overlay](https://github.com/napari/napari/blob/7b3f7ecfe0e74af5f52dd3eb380b5752b2f3378c/napari/components/viewer_model.py#L283) would do it - see for example [experimental](https://github.com/napari/napari/blob/7b3f7ecfe0e74af5f52dd3eb380b5752b2f3378c/napari/components/viewer_model.py#L547) which is [correctly documented as an attribute here](https://napari.org/dev/api/napari.Viewer.html)

**Comment by brisvag** [comment]
But can we do it in a way that shows the docstring of the `TextOverlay` model, rather than having to copy the docstring over to the property?

**Comment by melissawm** [comment]
This is related to #7221
